### DISCUSSION TITLE
Image inside Polygon

### DISCUSSION THREAD

**Discussion by GiovaniValdrighi** [visgl/deck.gl#5682] [Q&A]
I want to add a image inside a polygon of a GeoJson Layer, it's possible to fill the polygon with a solid color, and I already found the extension FillStyleExtension, but using it I could only fill the polygon with repetitions of the image (as an pattern). 
Is there a way to add a image in a polygon and stretch it to fit the polygon?

**Comment by sclebal** [comment]
I think maybe you have to extend the SolidPolygonLayer (example [here](https://github.com/visgl/deck.gl/issues/5021)) and modify the [fragment shader](https://github.com/visgl/deck.gl/blob/8.4-release/modules/layers/src/solid-polygon-layer/solid-polygon-layer-fragment.glsl.js) of that layer to allow you to set `gl_FragColor` with a [Texture2D](https://luma.gl/docs/api-reference/webgl/texture-2d). Maybe you can achieve it using inject with a hook like [this](https://deck.gl/docs/developer-guide/custom-layers/writing-shaders#fsmain-end) ([reference](https://deck.gl/docs/developer-guide/custom-layers/layer-extensions)).

A tricky solution might be to use a FillStyleExtension with a huge scale that the pattern doesn't need to be repeated.

Hope it helps you to begin with something!
